oshi-core-java25/src/main/java/oshi/ffm/windows/WinFFM.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2025 The OSHI Project Contributors
+ * SPDX-License-Identifier: MIT
+ */
+package oshi.ffm.windows;
+
+import java.lang.foreign.Arena;
+import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.Linker;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.SymbolLookup;
+import java.lang.invoke.MethodHandle;
+
+import static java.lang.foreign.ValueLayout.JAVA_CHAR;
+
+/**
+ * Utility class for working with the Foreign Function & Memory API (Java 24+). Provides helpers for library lookup,
+ * downcalls, and UTF-16 string conversion.
+ */
+public final class WinFFM {
+
+    private static final Linker LINKER = Linker.nativeLinker();
+
+    private WinFFM() {
+    }
```

Comment:
Linker is already defined in `ForeignFunctions` superclass with some general util functions.  You can put this in the middle.  Or, I don't see anything in here that's Windows specific except the utf16 method, which is still the same thing regardless of platform, so perhaps just add these methods to `ForeignFunctions`.

Consider renaming to WindowsForeignFunctions also.
```suggestion
public abstract class WinFFM extends ForeignFunctions {

    protected WinFFM() {
    }
```